c2corg_ui/static/js/documentservice.js

```diff
@@ -124,17 +124,23 @@ app.Document.prototype.hasAssociation = function(type, id) {
  * @param {appx.SimpleSearchDocument} doc
  * @param {string=} doctype Optional doctype
  * @param {boolean=} setOutingTitle
+ * @param {?boolean} editing
  * @export
  */
 app.Document.prototype.pushToAssociations = function(doc, doctype,
-    setOutingTitle) {
+    setOutingTitle, editing) {
   var associations = this.document.associations;
   doctype = doctype || app.utils.getDoctype(doc['type']);
   setOutingTitle = typeof setOutingTitle !== 'undefined' ?
     setOutingTitle : false;
   doc['new'] = true;
   associations[doctype].push(doc);
 
+  // when creating a route, make the first associated wp a main one
+  if (editing && associations.waypoints.length === 1) {
```

Comment:
In other words, it is the same behaviour when creating or editing - the first and only associated waypoint will be a main waypoint, that you can always unset. If you edit a route with for example one waypoint that is not a main one, it will have no impact untill you choose to click on "set as main waypoint" or delete this waypoint and add a new one -> which will become a main one, automatically.